File: internal/model/webhook.go
package model

import "time"

type Webhook struct {
	ID        int
	SessionID int
	URL       string
	Events    []string
	Enabled   bool
	Secret    string
	CreatedAt *time.Time
	UpdatedAt *time.Time
}

type WebhookEvent string

const (
	WebhookEventMessageReceived     WebhookEvent = "message.received"
	WebhookEventMessageSent         WebhookEvent = "message.sent"
	WebhookEventSessionConnected    WebhookEvent = "session.connected"
	WebhookEventSessionDisconnected WebhookEvent = "session.disconnected"
	WebhookEventSessionLoggedOut    WebhookEvent = "session.logged_out"
	WebhookEventAll                 WebhookEvent = "*"
)

File: painel/types/webhook.ts
export interface Webhook {
  id: string
  sessionId: string
  url: string
  events: string[]
  enabled: boolean
}

export interface WebhookFormData {
  url: string
  events: string[]
  enabled: boolean
}

export const WEBHOOK_EVENTS = [
  'message.received',
  'message.sent',
  'message.ack',
  'message.revoked',
  'presence.update',
  'group.join',
  'group.leave',
  'connection.update',
  'qr.update',
] as const

export type WebhookEvent = typeof WEBHOOK_EVENTS[number]
